src/merkle.rs

```diff
@@ -91,33 +92,39 @@ pub trait Element: Ord + Clone + AsRef<[u8]> + Sync + Send + Default + std::fmt:
 
 impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
     /// Creates new merkle from a sequence of hashes.
-    pub fn new<I: IntoIterator<Item = T>>(data: I) -> Result<MerkleTree<T, A, K>> {
-        Self::try_from_iter(data.into_iter().map(Ok))
+    pub fn new<I: IntoIterator<Item = T>>(data: I, branches: usize) -> Result<MerkleTree<T, A, K>> {
```

Comment:
I think we could make branches be a type parameter using typenum: https://docs.rs/typenum

Reply:
Very cool, thanks for pointing that out!  Updated.